propertyestimator/client.py

```diff
@@ -125,39 +143,34 @@ class Config:
         }
 
 
-class PropertyEstimator(object):
-    """The main object responsible for requesting a set of properties
-    be calculated by the low-level property calculation backend.
+class PropertyEstimatorClient:
+    """The :obj:`PropertyEstimatorClient` is the main object that users of the
+    property estimator will interface with. It is responsible for requesting
+    that a :obj:`PropertyEstimatorServer` estimates a set of physical properties,
+    as well as querying for when those properties have been estimated.
 
-    The property estimator acts as a client object, which is able to connect
-    and submit calculations to a remote (or even a local) property estimator server.
+    The :obj:`PropertyEstimatorClient` supports two main workflows: one where
+    a :obj:`PropertyEstimatorServer` lives on a remote supercomputing cluster
+    where all of the expensive calculations will be run, and one where
+    the users local machine acts as both the server and the client, and
+    all calculations will be performed locally.
 
     Examples
     --------
 
-    Setting up the server instance:
+    Setting up the client instance:
 
-    >>> # Create the backend which will be responsible for distributing the calculations
-    >>> from propertyestimator.backends import DaskLocalClusterBackend, PropertyEstimatorBackendResources
-    >>> calculation_backend = DaskLocalClusterBackend(1, 1, PropertyEstimatorBackendResources(1, 0))
-    >>>
-    >>> # Calculate the backend which will be responsible for storing and retrieving
-    >>> # the data from previous calculations
-    >>> from propertyestimator.storage import LocalFileStorage
-    >>> storage_backend = LocalFileStorage()
-    >>>
-    >>> # Create the server to which all calculations will be submitted
-    >>> from propertyestimator.server import PropertyCalculationRunner
-    >>> property_server = PropertyCalculationRunner(calculation_backend, storage_backend)
-    >>>
-    >>> # Instruct the server to listen for incoming submissions
-    >>> property_server.run_until_killed()
+    >>> from propertyestimator.client import PropertyEstimatorClient
+    >>> property_estimator = PropertyEstimatorClient()
 
-    If the server and client initialisation occur in the same function call (e.g. the same
-    main function), then `property_server.run_until_complete()` must be called after
-    all computations have been submitted.
+    If the :obj:`PropertyEstimatorServer` is not running on the local machine, you will
```

Comment:
If no `PropertyEstimatorServer` is found running locally or specified in the constructor, would it be possible for the `PropertyEstimator` (`PropertyEstimatorClient`) to create a `PropertyEstimatorServer` with simple `DaskLocalClusterBackend(ncpus)` and `LocalFileStorage()` compute and storage backends by default if none is specified?

Reply:
I've been struggling to think of a clean way to do this, but I think it should actually be possible - I think if we set up a PropertyEstimatorServer cli as an entrypoint, we could just `nohup` it from within python if no server address is given and no instance is running. We would then probably track the pid of the nohub, and make sure we kill it when we exit the method. The only issue with this is that if something goes wrong and the program doesn't exit cleanly, the server would be left running perhaps without the users knowledge..

Reply:
The only other issue that may pop up (and I don't know if this would ever actually happen) might be that you call this method, and it spins up a local server. You may then call this method again from another script on the same machine. This new script would see the spun up server and launch its jobs on there. The original script may then finish and kill the server, before the jobs submitted by the second script have finished.

Reply:
That could be resolved by simply having two behaviors:
* If no server is specified, spin up a local server that the client owns (and kills when the client dies)
* If a server is specified, use that (but don't kill it when the client dies)

Reply:
Another solution could be adding a flag to the `PropertyEstimatorServer` that makes it self-kill after a period of inactivity and have the `PropertyEstimatorClient` always turn on that flag when automatically creating a server.